src/query_router.rs

```diff
@@ -150,9 +151,13 @@ impl QueryRouter {
                     || self.pool_settings.sharding_key_regex.is_some()
                 {
                     // Check only the first block of bytes configured by the pool settings
-                    let len = message_cursor.get_i32() as usize;
                     let seg = cmp::min(len - 5, self.pool_settings.regex_search_limit);
```

Comment:
is this variable unused now?

Reply:
it's used in the seg calculation. It needs to be read off the cursor to progress the index that it's pointing to